src/util/positioning.ts

```diff
@@ -0,0 +1,102 @@
+import {ElementRef} from '@angular/core';
+
+export interface ElementPosition {
+  width: number;
+  height: number;
+  top: number;
+  left: number;
+}
+
+// previous version:
+// https://github.com/angular-ui/bootstrap/blob/07c31d0731f7cb068a1932b8e01d2312b796b4ec/src/position/position.js
+export class Positioning {
+  private getStyle(element: HTMLElement, prop: string): string { return window.getComputedStyle(element)[prop]; }
+
+  private isStaticPositioned(element: HTMLElement): boolean {
+    return (this.getStyle(element, 'position') || 'static') === 'static';
+  }
+
+  private parentOffsetElement(element: HTMLElement): HTMLElement {
+    let offsetParentEl: HTMLElement = element.parentElement || document.documentElement;
+    while (offsetParentEl && offsetParentEl !== document.documentElement) {
+      offsetParentEl = offsetParentEl.offsetParent as HTMLElement;
+    }
+    return offsetParentEl || document.documentElement;
+  }
+
+  position(element: HTMLElement): ElementPosition {
+    let elOffset: ElementPosition = this.offset(element);
+    let elBCR: ClientRect = element.getBoundingClientRect();
+    let parentOffset: ElementPosition = {width: 0, height: 0, top: 0, left: 0};
+    let offsetParentEl: HTMLElement = this.parentOffsetElement(element);
+
+    if (offsetParentEl !== document.documentElement) {
+      parentOffset = this.offset(offsetParentEl);
+      parentOffset.top += offsetParentEl.clientTop - offsetParentEl.scrollTop;
+      parentOffset.left += offsetParentEl.clientLeft - offsetParentEl.scrollLeft;
+    }
+
+    return <ElementPosition>{
```

Comment:
the compiler will check that if the function declares a return type, that the returned type matches what's declared.